tests/unit/dataset/generator/test_prompt_generator.py

```
@@ -178,6 +178,48 @@ def testgenerate_prompt_large_number(self, basic_config):
+        """generate_prompt trims and retries when re-encode returns too many tokens."""
+        tokenizer, prompts, prefix_prompts = basic_config
+
+        # First encode call returns 2 extra tokens; subsequent calls are accurate.
+        real_encode = tokenizer._mock_encode
+        call_count = {"n": 0}
+
+        def drifting_encode(text, **kwargs):
+            call_count["n"] += 1
+            ids = real_encode(text, **kwargs)
+            if call_count["n"] == 1:
+                return ids + [998, 999]
```

Ruff (0.15.21)

[warning] Consider `[*ids, 998, 999]` instead of concatenation

Replace with `[*ids, 998, 999]`

(RUF005)